library/core/src/main/java/com/google/android/exoplayer2/text/webvtt/WebvttParserUtil.java

```diff
@@ -52,8 +54,16 @@ public static void validateWebvttHeaderLine(ParsableByteArray input) throws Pars
    * @param input The input from which the line should be read.
    */
   public static boolean isWebvttHeaderLine(ParsableByteArray input) {
-    @Nullable String line = input.readLine();
-    return line != null && line.startsWith(WEBVTT_HEADER);
+    @Nullable  String line = input.readLine();
+    boolean isValidHeader = line != null && line.startsWith(WEBVTT_HEADER);
+    if (isValidHeader) {
+      // According to w3c spec, TIMESTAMP-MAP header can be on same line as WEBVTT header.
+      int headerStartAt = line.indexOf(TIMESTAMP_MAP_HEADER);
+      if (headerStartAt != -1) {
+        input.setPosition(headerStartAt);   // set so next readLine() reads TIMESTAMP header
+      }
+    }
+    return isValidHeader;
```

Comment:
It seemed best to keep all the logic dealing with the WEBVTT_HEADER here, Pantos is ambiguous as to termination, the w3c spec explicitly says it can be followed by a space or tab.   This code handles space, tab or newline termination.